ext/signals.c

```diff
@@ -102,6 +107,138 @@ static bool ddtrace_crashtracker_check_result(ddog_VoidResult result, const char
+    zend_string *value = zend_ini_get_value(ini_zstr);
+    zend_string_release(ini_zstr);
+    return value;
+}
+
+// Pass in a key like "php.opcache.enable" and "php." will get stripped off,
+// and that's what the INI setting will be.
+static void ddtrace_crasht_add_opcache_tag(
+    ddog_Vec_Tag *tags,
+    ddog_CharSlice key
+) {
+    const ddog_CharSlice PREFIX = DDOG_CHARSLICE_C("php.");
+    ZEND_ASSERT(key.len > PREFIX.len && memcmp(key.ptr, PREFIX.ptr, PREFIX.len) == 0);
+
+    ddog_CharSlice ini = {
+        .ptr = key.ptr + PREFIX.len,
+        .len = key.len - PREFIX.len,
+    };
+    zend_string *value = ddtrace_crasht_ini_get_value(ini);
+    // On PHP 8.0+ these INI should all exist, but guard against the NULL case
+    // in case something goes wrong, or this changes in a future version.
+    ddog_CharSlice val = DDOG_CHARSLICE_C("[not found]");
+    if (value) {
+        val.ptr = ZSTR_VAL(value);
+        val.len = ZSTR_LEN(value);
+    }
+    ddog_Vec_Tag_PushResult result = ddog_Vec_Tag_push(tags, key, val);
+    if (UNEXPECTED(result.tag != DDOG_VEC_TAG_PUSH_RESULT_OK)) {
+        ddtrace_crasht_failed_tag_push(&result.err, key);
+    }
+}
+
+static zend_extension *ddtrace_crasht_find_zend_extension(const char *name) {
+    //if (maybe_opcache->name && strcmp(maybe_opcache->name, "Zend OPcache") == 0) {
+    const zend_llist *list = &zend_extensions;
+    zend_extension *maybe_opcache = NULL;
+    for (const zend_llist_element *item = list->head; item; item = item->next) {
+        maybe_opcache = (zend_extension *)item->data;
+        if (maybe_opcache->name && strcmp(maybe_opcache->name, name) == 0) {
+            return maybe_opcache;
+        }
+    }
+    return NULL;
+}
+#endif
+
+// Fetches certain opcache tags and adds them with the pattern of php.opcache.*.
+static void ddtrace_crasht_add_opcache_tags(ddog_Vec_Tag *tags) {
+#if PHP_VERSION_ID >= 80000
+    // If opcache isn't loaded, don't attach any opcache tags.
+    if (!ddtrace_crasht_find_zend_extension("Zend OPcache")) {
+        return;
+    }
+
+    // opcache.jit_buffer_size is INI_SYSTEM, so we can check it now. If it's
+    // zero, then JIT won't operate.
+    ddog_CharSlice jit_buffer_size = DDOG_CHARSLICE_C("[not found]");
+    {
+        ddog_CharSlice ini = DDOG_CHARSLICE_C("opcache.jit_buffer_size");
+        zend_string *value = ddtrace_crasht_ini_get_value(ini);
+        if (!value) {
+            return;
+        }
+
+        // Parse the quantity similarly to OnUpdateLong.
+#if PHP_VERSION_ID >= 80200
+        zend_string *errstr = NULL;
+        zend_long quantity = zend_ini_parse_quantity(value, &errstr);
+        if (errstr) zend_string_release(errstr);
+#else
+        zend_long quantity = zend_atol(ZSTR_VAL(value), ZSTR_LEN(value));
+#endif
+        if (quantity <= 0) {
+            return;
+        }
+        jit_buffer_size.ptr = ZSTR_VAL(value);
+        jit_buffer_size.len = ZSTR_LEN(value);
```

Comment:
```suggestion
        jit_buffer_size = dd_zend_string_to_CharSlice(value);
```
nit (same for enable_cli)